os_stub/cryptlib_openssl/pk/x509.c

```diff
@@ -2618,17 +2623,26 @@ bool libspdm_gen_x509_csr(size_t hash_nid, size_t asym_nid,
         goto free_all;
     }
 
-    basic_constraints_ext = X509V3_EXT_conf_nid(
-        NULL, NULL, NID_basic_constraints,
-        is_ca ? "CA:TRUE" : "CA:FALSE");
-    if (!basic_constraints_ext) {
-        sk_X509_EXTENSION_free(exts);
-        ret = 0;
-        goto free_all;
+    /*set basicConstraints*/
+    if (base_cert != NULL) {
+        num_exts = X509_get_ext_count(base_cert);
+
+        for (int i = 0; i < num_exts; i++) {
+            sk_X509_EXTENSION_push(exts, X509_get_ext(base_cert, i));
+        }
+    } else {
```

Comment:
I have a question: is it consistent for the `basic_constraints` between `base_cert` and `is_ca`?

Reply:
Are you worried about not appending a `CA:TRUE`/`CA:FALSE`?

I think it would be up to the caller to make sure the constraints supplied are correct